core-bundle/src/Twig/Inspector/InspectorNodeVisitor.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\Twig\Inspector;
+
+use Contao\CoreBundle\Twig\Slots\SlotNode;
+use Psr\Cache\CacheItemPoolInterface;
+use Twig\Environment;
+use Twig\Node\Expression\ConstantExpression;
+use Twig\Node\ModuleNode;
+use Twig\Node\Node;
+use Twig\NodeVisitor\NodeVisitorInterface;
+
+/**
+ * @experimental
+ */
+final class InspectorNodeVisitor implements NodeVisitorInterface
```

Comment:
I think I would also try to extract the entire slots management in Twig from this PR because I don't see why we couldn't use this in a different context. I mean, at least if we get to the stage of where we think this is going in the right direction, it would make the PR a lot easier to review if that part were already merged before.